cmd/snap/cmd_get.go

```diff
@@ -96,9 +148,33 @@ func (x *cmdGet) Execute(args []string) error {
 		return err
 	}
 
+	isTerminal := terminal.IsTerminal(int(os.Stdin.Fd()))
+
 	var confToPrint interface{} = conf
-	if !x.Document && len(confKeys) == 1 {
-		confToPrint = conf[confKeys[0]]
+	if !x.Document && !x.List && len(confKeys) == 1 {
+		// if single key was requested, then just output the value unless it's a map,
+		// in which case it will be printed as a list below.
+		if _, ok := conf[confKeys[0]].(map[string]interface{}); !ok {
+			confToPrint = conf[confKeys[0]]
+		}
+	}
+
+	if cfg, ok := confToPrint.(map[string]interface{}); ok && !x.Document {
+		// TODO: remove this conditional and the warning below after a transition period.
+		if isTerminal || x.List {
+			w := tabWriter()
+			defer w.Flush()
+
+			rootRequested := len(confKeys) == 0
+			fmt.Fprintf(w, "Key\tValue\n")
+			values := flattenConfig(cfg, rootRequested)
+			for _, v := range values {
+				fmt.Fprintf(w, "%s\t%v\n", v.path, v.value)
+			}
+			return nil
+		} else {
+			fmt.Fprintf(Stderr, i18n.G(`WARNING: The output of "snap get" will become a list with columns - use -d to force JSON output.\n`))
```

Comment:
Instead of _"use -d to force JSON output"_, we can say _"use -d or -l to force the output format",_ so it's clear that there's a choice. The developer might be trying to use a list, since that's what is being seen in the terminal.